integration-tests/jbang/src/test/java/io/quarkus/jbang/RunJBangTest.java

```diff
@@ -0,0 +1,130 @@
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.zeroturnaround.exec.ProcessExecutor;
+import org.zeroturnaround.exec.ProcessResult;
+import org.zeroturnaround.exec.StartedProcess;
+import org.zeroturnaround.exec.stream.LogOutputStream;
+
+class RunJBangTest {
+
+    private static final String LATEST_DOWNLOAD_JBANG_ZIP = "https://www.jbang.dev/releases/latest/download/jbang.zip";
+    private static final File JBANG_FILE = new File("target/test-classes/devmode.java");
+    private static Path JBANG_DIR = null;
+
+    int port;
+
+    /**
+     * Downloads latest jbang in static dir and setup JBANG_DIR to be 100% isolated
+     * from whatever jbang could otherwise be installed on the system.
+     */
+    @BeforeAll
+    static void downloadLatestJBang(@TempDir Path tempdir) throws IOException {
+        InputStream in = new URL(LATEST_DOWNLOAD_JBANG_ZIP).openStream();
+        JBANG_DIR = tempdir.resolve(".jbang");
+        Path zipfile = tempdir.resolve("jbang.zip");
+        Files.copy(in, zipfile, StandardCopyOption.REPLACE_EXISTING);
+        TestHelper.unzip(zipfile, JBANG_DIR, true, null);
+    }
+
+    @BeforeEach
+    void init() {
+        port = TestHelper.getRandomNumber(8000, 9999);
+    }
+
+    private ProcessExecutor getJBangExecutor(String... args) {
+        String cmd;
+        if (isWindows()) {
+            cmd = "bin/jbang.cmd";
+        } else {
+            cmd = "bin/jbang";
+        }
+
+        List<String> realargs = new ArrayList<>();
+        realargs.add(JBANG_DIR.resolve(cmd).toString());
+        realargs.addAll(Arrays.asList(args));
+
+        return new ProcessExecutor().command(realargs)
+                .environment("JBANG_DIR", JBANG_DIR.toAbsolutePath().toString())
+                .redirectErrorStream(true).readOutput(true);
+    }
+
+    /**
+     * just a sanity check that jbang is behaving as we expect with respect to config
+     **/
+    @Test
+    void testRunJBangWithProperDir() throws IOException, InterruptedException, TimeoutException {
+
+        String result = getJBangExecutor("version", "--verbose")
+                .exitValue(0)
+                .execute().outputUTF8();
+        assertThat(result).contains(JBANG_DIR.toAbsolutePath().toString());
+    }
+
+    @Test
+    void testRunJBangInDevMode() throws IOException, InterruptedException, TimeoutException, ExecutionException {
+
+        final CountDownLatch listening = new CountDownLatch(1);
+
+        StartedProcess exec = getJBangExecutor("-Dquarkus.dev", "-Dq.v=999-SNAPSHOT",
```

Comment:
@gsmet I know in past there were issues with tests relying on 999-SNAPSHOT which fails 
on release build...what is the proper way to get the release version to use during the test ?

Reply:
I have no idea what this version is used for and what how q.v would be consumed so I can't really answer.

Reply:
The version of quarkus that is being built. 

You had examples in past where you complained some maven/gradle tests failed during release since at that point it's 2.0.0.Alpha2 or similar. 

So just wondering if you know how that was solved as I couldn't locate it yet.

Reply:
@gsmet does that info help ? i'm simply just trying to get the current version of quarkus being built.

@aloubyansky maybe you have an idea?

the test is a unit test, running during test of the build and I need to have  jbang executed and use the currently building quarkus artifacts which wont be 999-snapshot at release time.

Reply:
Sure, usually we pass system property to the surefire/failsafe plugin that is read in the test.